### PR TITLE
fix: npmcli run-script separated from bundle as external dep

### PR DESCRIPTION
**Pull request by derberg**
**Description**

- ok, I think I need to improve my reading skills, `--external` in ncc means that it is not going to bundle package from different location, it means that it will not bundle it at all and requires it to be stored separately. So basically `run-script` is a package we need to store in the repo in `node_modules`. 

**Related issue(s)**
Fixes https://github.com/asyncapi/github-action-for-generator/issues/111

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,10 @@
...
-node_modules
+node_modules/*
 .DS_Store
 output
+
+#we must commit this single package to the repo as ncc cannot bundle it and the dist bundle is configure to use run-script as external package
+#yes, below looks strange but this is how it is done with git
+!node_modules/@npmcli/
+node_modules/@npmcli/*
+!node_modules/@npmcli/run-script/
```
Comment (magicmatatjahu) [inline .gitignore:10]: Is `!node_modules/@npmcli/run-script/` not enough?

Comment (derberg) [reply .gitignore:10]: you didn't read my comment few lines above, did you? 😋

Comment (magicmatatjahu) [reply .gitignore:10]: I read and that I really wonder that need this construct 😄

**Review by magicmatatjahu — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (derberg) [reply .gitignore]: I've added [this](https://stackoverflow.com/questions/5533050/gitignore-exclude-folder-but-include-specific-subfolder) link to the comment

**Review by magicmatatjahu — DISMISSED** [review]

**Comment by derberg** [comment]
@magicmatatjahu please approve again, I had to commit typescript to node_modules too. I tested and also @pioneer2k confirmed it works now.

**Comment by derberg** [comment]
@magicmatatjahu but @pioneer2k can use the PR now in his pipelines for few days and you definitely should not approve it this week while being on holidays. Any attempt to approve will be dismissed :trollface:

**Review by magicmatatjahu — APPROVED** [review]
LGTM, but in the future: please divide the large PR into several smaller ones.

**Comment by derberg** [comment]
@magicmatatjahu I don't think it is possible with bug fixes and when node_modules are pushed to the PR, but I will definitely do my best to do it. Thanks a lot @magicmatatjahu 🤝

**Pull request merged**